app/controllers/api/v1/locations_controller.rb

```diff
@@ -25,6 +25,12 @@ def create
     end
```

Comment:
Your authetnicate_user method (at the bottom of this file) needs to be updated. we cant user flash or redirect when in an API::V1 controller. instead, your should return a json message that says they do not have access to the page. Also set the status message to 404

this is a little out of scope of the PR mind you, so I will allow to table this as long as your next story fixes this issue in any API controllers. otherwise, just update it now.